spinalcordtoolbox/registration/register.py

```diff
@@ -29,6 +29,7 @@
 import spinalcordtoolbox.image as image
 from spinalcordtoolbox.math import laplacian
 from spinalcordtoolbox.registration.landmarks import register_landmarks
+from spinalcordtoolbox.scripts import sct_resample, sct_register_multimodal
```

Comment:
> There are still a lot of checks failing (maybe weirdly even more than before) but from what I see and understand it doesn't seem to be necessarily linked to the modifications done in this PR, so I don't really know how this should be resolved. What do you think ?

Ahhhh, I see the problem. This line has caused a [`circular import`](https://stackoverflow.com/q/22187279), In other words:

* `register.py` imports `sct_register_multimodal`
* But, `sct_register_multimodal` imports functions from `register.py`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/b12fe5d3b615431355dd01a845fe596ea5f9dbba/spinalcordtoolbox/scripts/sct_register_multimodal.py#L41

There are 2 fixes we could apply here:

1. Try to not use `sct_register_multimodal`, but instead try to pre-register the images using the `register_wrapper()` function directly.
    * The problem with this is that right now, `register_wrapper()` is *also* in a CLI script right now (`sct_register_to_template`), so I fear that this would also cause a circular import error (since, `register.py` would import from `sct_register_to_template`, and `sct_register_to_tempalte` would import from `register.py`).
    * So, I think to solve the root cause, we would need to perform a deeper refactor to better separate the CLI/API. 
    * But, I think we can save this until after we finish up this PR.
2. The alternative is to apply the workaround from [this SO post](https://stackoverflow.com/a/22210807):

	  > [Circular imports] can work just fine, if you set them up correctly.
	  > 	  
	  > The easiest way to do so is to use import `my_module` syntax, rather than `from my_module import some_object`. The former will almost always work, even if `my_module` included imports us back. The latter only works if `my_object` is already defined in `my_module`, which in a circular import may not be the case.

I'm going to try and add a commit that does this to see if I can fix the issue.

Reply:
Even after fixing the first circular import, it looks like there are even more circular imports.

Perhaps tackling 1. now *is* the best option, rather than putting it off.

Reply:
Update: Moving `register_wrapper` out of `sct_register_to_template` is a lot harder than I anticipated, because `register_wrapper` is designed to use a `Param` object that gets defined inside of `sct_register_to_template`. 

But, I think I have a workable solution here...

Reply:
> Ahhhh, I see the problem. The change from https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/7e43cf8d90b7557c03257e7d1088497b378f20c8 has caused a [circular import](https://stackoverflow.com/q/22187279)

Oh yes you're right, good catch !

> Okay! I've opened https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3811, which moves register_wrapper() out of sct_register_to_template. This allows us to use register_wrapper() to perform the -identity step, rather than sct_register_multimodal, which will fix the circular import error.

Ok perfect, that's really cool thank you ;)

> To confirm that the tests all pass, I ran [this GitHub Actions run](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/2511107436).

Looks nice !!

> Once https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3811 is merged, we can remake the PR using the new branch.

Alright ! Thank you very much for looking into this in depth :)

Reply:
#3811 has been merged, so I've done two things:

* Rebased this PR onto master (which incorporates the changes from [#3811](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3811)).
* Added commit [1dfe5dc](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3807/commits/1dfe5dc80beef565e77d8ea6495d23cf73e8c4b2), which was originally from my test branch.

This PR should now be ready for a proper review!